pkg/cli/tail.go

```diff
@@ -222,24 +274,51 @@ func (c *TailCommand) Run(ctx context.Context, args []string) error {
 	return nil
 }
 
-func (c *TailCommand) tail(ctx context.Context) ([]*loggingpb.LogEntry, error) {
-	var p logPuller
-	if c.testPuller != nil {
-		p = c.testPuller
-	} else {
-		logClient, err := logging.NewClient(ctx)
+func (c *TailCommand) stream(ctx context.Context, extra []validation.Validator, puller logPuller) error {
+	// logCh is used to receive logEntriesfrom StreamPull
+	logCh := make(chan *loggingpb.LogEntry)
+
+	var failCount, totalCount int
+
+	var perr error
+
+	go func() {
+		if err := puller.StreamPull(ctx, c.queryFilter(), logCh); err != nil {
+			perr = fmt.Errorf("StreamPull failed: %w", err)
+		}
+		close(logCh)
+	}()
```

Comment:
```suggestion
	go func() {
		defer close(logCh)
		if err := puller.StreamPull(ctx, c.queryFilter(), logCh); err != nil {
			perr = fmt.Errorf("StreamPull failed: %w", err)
		}
	}()
```

Reply:
Yes, I missed this one. Even on error, we should close the channel